src/parser/HLSTree.cpp

```diff
@@ -101,35 +101,21 @@ ContainerType DetectContainerTypeFromExt(std::string_view extension)
     return ContainerType::INVALID;
 }
 
-// \brief Workaround to get audio codec from CODECS attribute list
+// \brief Get the first audio codec string from CODECS attribute list
 std::string GetAudioCodec(std::string_view codecs)
 {
-  //! @todo: this way to get the audio codec is inappropriate and can lead to bad playback
-  //! this because CODECS attribute its optionals and not guarantee to provide full codec list
-  //! the codec format should be provided by MP4 demuxer
-
-  // The codec search must follow exactly the following order, this is currently the best workaround
-  // to make multi-channel audio formats work, but since CODECS attribute is unreliable
-  // this workaround can still cause playback problems
-  if (codecs.find(CODEC::FOURCC_EC_3) != std::string::npos)
-    return CODEC::FOURCC_EC_3;
-  else if (codecs.find(CODEC::FOURCC_AC_3) != std::string::npos)
-    return CODEC::FOURCC_AC_3;
-  else
-    return CODEC::FOURCC_MP4A;
-}
-// \brief Workaround to get audio codec from representation codecs list
-std::string GetAudioCodec(const PLAYLIST::CRepresentation* repr)
-{
-  const auto& codecs = repr->GetCodecs();
-  if (CODEC::Contains(codecs, CODEC::FOURCC_EC_3))
-    return CODEC::FOURCC_EC_3;
-  else if (CODEC::Contains(codecs, CODEC::FOURCC_AC_3))
-    return CODEC::FOURCC_AC_3;
-  else
-    return CODEC::FOURCC_MP4A;
+  // Some manifests can provide CODECS attribute with more audio codecs
+  // it seems that usually the first in order of apparence is the one referring to the specified
+  // AUDIO group attribute, so we return the first one found
```

Comment:
Great find here. To me this was always a problem that would be difficult to solve but if they're ordered then our lives are much easier :)

Reply:
this can works for mp4 case with group attribute,
but im quite sure that this is still wrong for the case when a video TS have embedded/included multiple audio streams (without audio group)
also im not sure that currently the parser is able to handle TS stream with more included audio streams
it seems to me that it is not even considered, more likely if/when we will find a sample stream to make tests will require many more code adjustments

Reply:
Yes TS case is a different baby. Same goes for fMP4 with multiple tracks (audio/video, audio/audio or video/audio/audio)... we can add to the long list of wishes! :)